sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/TopicSubscriptionWithRuleOperationsSample.java

```diff
@@ -0,0 +1,222 @@
+
+    static final String SERVICE_BUS_CONNECTION_STRING = "{Your Service Bus Connection String}";
+    static final String SERVICE_BUS_TOPIC_NAME = "{Your Service Bus Topic Name}";
+    static final String ALL_MESSAGES_SUBSCRIPTION_NAME = "{Subscription 1 Name}";
+    static final String SQL_FILTER_ONLY_SUBSCRIPTION_NAME = "{Subscription 2 Name}";
+    static final String SQL_FILTER_WITH_ACTION_SUBSCRIPTION_NAME = "{Subscription 3 Name}";
+    static final String CORRELATION_FILTER_SUBSCRIPTION_NAME = "{Subscription 4 Name}";
+    static final String DEFAULT_SUBSCRIPTION_RULE_NAME = "$Default";
+    static final String SQL_FILTER_ONLY_SUBSCRIPTION_RULE_NAME = "RedSqlRule";
+    static final String SQL_FILTER_WITH_ACTION_SUBSCRIPTION_RULE_NAME = "BlueSqlRule";
+    static final String CORRELATION_FILTER_SUBSCRIPTION_RULE_NAME = "ImportantCorrelationRule";
+    static final String MESSAGE_JSON_ITEM_NAME_LABEL = "label";
+    static final String MESSAGE_JSON_ITEM_NAME_CORRELATION_ID = "correlationId";
+    static final Gson GSON = new Gson();
+
+    static ServiceBusSenderAsyncClient topicSenderAsyncClient;
+    static ServiceBusAdministrationAsyncClient administrationAsyncClient;
+
+    /**
+     * Main method to invoke this demo on how to receive {@link ServiceBusReceivedMessage messages}
+     * by topic subscriptions from an Azure Service Bus Topic.
+     *
+     * @param args Unused arguments to the program.
+     * @throws InterruptedException If the program is unable to sleep while waiting for the receive to complete.
+     */
+    public static void main(String[] args) throws InterruptedException {
+        TopicSubscriptionWithRuleOperationsSample app = new TopicSubscriptionWithRuleOperationsSample();
+        app.run();
+    }
+
+    /**
+     * This method to invoke this demo on how to receive {@link ServiceBusReceivedMessage messages}
+     * by topic subscriptions from an Azure Service Bus Topic.
+     *
+     * @throws InterruptedException If the program is unable to sleep while waiting for the receive to complete.
+     */
+    void run() throws InterruptedException {
+
+        topicSenderAsyncClient = new ServiceBusClientBuilder()
+            .connectionString(SERVICE_BUS_CONNECTION_STRING)
+            .sender()
+            .topicName(SERVICE_BUS_TOPIC_NAME)
+            .buildAsyncClient();
+
+        administrationAsyncClient = new ServiceBusAdministrationClientBuilder()
+            .connectionString(SERVICE_BUS_CONNECTION_STRING)
+            .serviceVersion(ServiceBusServiceVersion.getLatest())
+            .buildAsyncClient();
+
+        // First Subscription is already created with default rule. Leave as is.
+        System.out.println(String.format("SubscriptionName: %s, Removing and re-adding Default Rule", ALL_MESSAGES_SUBSCRIPTION_NAME));
+        administrationAsyncClient.deleteRuleWithResponse(SERVICE_BUS_TOPIC_NAME, ALL_MESSAGES_SUBSCRIPTION_NAME, DEFAULT_SUBSCRIPTION_RULE_NAME).block();
```

Comment:
Use `deleteRule` instead of `deleteRuleWithResponse` since you don't care the response.

Reply:
According to your comment, fixed in new version.